### PR TITLE
Make the toggle collision work

### PR DESCRIPTION
**Pull request by Lyght-1**
-fix the collision bug
-fix getting stuck on a wall for a sec even with collision off.

### CODE REVIEW

File: Extensions/PlayerControl_Ext.cs [hunk 1]
```diff
@@ -101,7 +101,10 @@ public void PracticeFKeys(PlayerControl __instance)
 				MainManager_Ext.toggleCollision = !MainManager_Ext.toggleCollision;
 				__instance.entity.detect.enabled = !MainManager_Ext.toggleCollision;
 				__instance.entity.ccol.enabled = !MainManager_Ext.toggleCollision;
+				__instance.entity.feet.enabled = !MainManager_Ext.toggleCollision;
 				MainManager_Ext.toggleInfJump = MainManager_Ext.toggleCollision;
+				if (MainManager_Ext.toggleCollision)
+					__instance.entity.hitwall = false;
```
Comment (aldelaro5) [inline Extensions/PlayerControl_Ext.cs]: indentation

File: Patches/EntityControl.cs [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
     {
-      if (!MainManager_Ext.toggleCollision && __instance.name == "Player 0")
-        return true;
-
-      return false;
+        static void Postfix(EntityControl __instance)
+        {
+            if (__instance.playerentity)
+                __instance.ccol.enabled = !MainManager_Ext.toggleCollision;
+        }
```
Comment (aldelaro5) [inline Patches/EntityControl.cs]: Indentation changed, pls keep it to 2 spaces.

File: Patches/GroundDetector.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+using System;
+
+namespace SpeedrunPractice.Patches
+{
+    [HarmonyPatch(typeof(GroundDetector), "OnTriggerStay")]
+    public class PatchGroundDetectorOnTriggerStay
+    {
+        static bool Prefix(GroundDetector __instance) => __instance.enabled;
+    }
+}
```
Comment (aldelaro5) [inline Patches/GroundDetector.cs:12]: Indentation.

File: Patches/RayDetector.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+using System;
+
+namespace SpeedrunPractice.Patches
+{
+    [HarmonyPatch(typeof(RayDetector), "OnTriggerStay")]
+    public class PatchRayDetectorOnTriggerStay
+    {
+        static bool Prefix(RayDetector __instance) => !MainManager_Ext.toggleCollision;
+    }
+}
```
Comment (aldelaro5) [inline Patches/RayDetector.cs:12]: Indentation

**Review by aldelaro5 — CHANGES_REQUESTED** [review]
New files should be indented by 2 spaces and existing file changes had indentation change which I would like to see reverted.

Other than that LGTM, but like I mentioned with the collision viewer PR, I would need to test this, though this should be approved relatively quickly.

**Comment by aldelaro5** [comment]
Hey, just a reminder if it is possible to address my code reviews (and answer question cause there was one line that was confusing me) because I expect to actually come back to this in a few days so if the code reviews are addressed by the time I get to testing, it will make it quick.

Due to how urgent 2 of your prs are, I might be forced to merge as is when I confirm my tests, but keep in mind that I will have to fix the issues myself which isn't what you would do normally (you are supposed to address the issues yourself before I merge). What this means is that although it might pass for these prs because they are urgent, I am normally am not going to merge until the code reviews are addressed which is going to block further prs. Just to let you know.

**Comment by aldelaro5** [comment]
Last call: if none of my code reviews are addressed by Saturday after my tests, I am forced to preemptively merge and take over the PRs. Since 2 of your PRs will generate I conflict, I will have to coauthor one of them (normally I would ask permissions, but it's urgent enough to do it if I am forced to).

In the event that you can't address these reviews, please know that I won't do this next time. If I post a code review, pls address it before we can move on.

**Comment by Lyght-1** [comment]
Should be good now ?

**Review by aldelaro5 — APPROVED** [review]
LGTM, though I kinda want to check why that prefix on the grounDetector was needed because in theory, this event can't fire if the existing monoBehaviour is disabled, but this is urgent to merge so it will have to do for now

**Comment by aldelaro5** [comment]
> Should be good now ?

yeah, I just made a release, might check your new pr now though because it seems urgent.

Btw, was the GroundDetector patch needed? Unity shouldn't fire events on disabled objects

**Comment by Lyght-1** [comment]
Yes it was. OnTrigger events are still called even if the object is disabled.

**Comment by aldelaro5** [comment]
huh 
``` Note: Trigger events are only sent if one of the colliders also has a rigidbody attached. Trigger events are sent to disabled MonoBehaviours to allow enabling Behaviours in response to collisions.```
https://docs.unity3d.com/ScriptReference/MonoBehaviour.OnTriggerStay.html

I didn't know that!

**Pull request merged**